### PR TITLE
fix aes-xts bug on aarch64 big-endian env.

### LINKED ISSUE #20796: aarch64 big-endian AES-XTS bug
I got a error when I run test on a aarch64 big-endian env，it's appear to be a bug of aarch64 assembly optimization for AES-XTS.
The specific test is `make TESTS='test_evp' test`.

### PR DESCRIPTION
**Pull request by Liu-ErMeng**
fix #20796 

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
@tom-cosgrove-arm Could you please review this?

File: test/recipes/30-test_evp_data/evpciph_aes_common.txt [hunk 1]
```diff
@@ -1259,6 +1259,19 @@ IV = 9a785634120000000000000000000000
 Plaintext = 000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f303132333435363738393a3b3c3d3e3f404142434445464748494a4b4c4d4e4f5051
 Ciphertext = edbf9dace45d6f6a7306e64be5dd824b2538f5724fcf24249ac111ab45ad39233ad6183c66fa548a3cdf3e36d2b21ccdc6bc657cb3aeb87ba2c5f58ffafacd765ecc4c85c0a01bf317b823fbd6111956d0a0
 
+# To cover the branchs of assembly code of aes_v8_xts_encrypt(decrypt)
```
Comment (tom-cosgrove-arm) [inline test/recipes/30-test_evp_data/evpciph_aes_common.txt:1262]: (minor) it's "branches" not "branchs"

Comment (Liu-ErMeng) [reply test/recipes/30-test_evp_data/evpciph_aes_common.txt]: fixed. Sorry for my poor English.

File: test/recipes/30-test_evp_data/evpciph_aes_common.txt [hunk 2]
```diff
@@ -1259,6 +1259,19 @@ IV = 9a785634120000000000000000000000
 Plaintext = 000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f303132333435363738393a3b3c3d3e3f404142434445464748494a4b4c4d4e4f5051
 Ciphertext = edbf9dace45d6f6a7306e64be5dd824b2538f5724fcf24249ac111ab45ad39233ad6183c66fa548a3cdf3e36d2b21ccdc6bc657cb3aeb87ba2c5f58ffafacd765ecc4c85c0a01bf317b823fbd6111956d0a0
 
+# To cover the branches of assembly code of aes_v8_xts_encrypt(decrypt)
```
Comment (xkqian) [inline test/recipes/30-test_evp_data/evpciph_aes_common.txt:1262]: Where does the data come from? are they from specs? or are generated by oneself?

Comment (Liu-ErMeng) [reply test/recipes/30-test_evp_data/evpciph_aes_common.txt:1262]: It's come from [line 1029](https://github.com/openssl/openssl/pull/20797/files/eeaf34baac3a42ca28eaddea0b7d983163788b17#diff-556bad09eca86c009d60c31a75480597ed6d89aa40416c97dc550d968571cd7bR1029). It's just the same but with half length of input/output to cover the branches of the assembly code.

**Comment by xkqian** [comment]
Cause we have so many modes such as ecb, cbc, ctr, xts, etc. modes to use the key schedule, If we want to re-structure the code, maybe it's better to change in the key schedule(init step) to store the round keys by byte, and then change the keys load in this file relatively. 
Anyway, I just learned that there were some patches to fix the big-endian issue for other modes, and have no need to waste extra effort to change them back. 

I just have tested the PR locally with big-endian and little-endian, all of them passed.
The change itself seems no problem, which change the load instructions to align with the store instructions of the schedule keys.
Currently agree that we can change the endian order in the AES bodys,  just as this PR did.

**Review by tom-cosgrove-arm — APPROVED** [review]
(reviewed internally and approved by Xiaokang - @xkqian)

**Comment by xkqian** [comment]
> (reviewed internally and approved by Xiaokang - @xkqian - thanks)

You're welcome.

**Review by xkqian — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master, 3.1, and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**